keyvalue/server/pom.xml

```diff
@@ -46,7 +46,12 @@
     </dependency>
     <dependency>
       <groupId>org.alluxio</groupId>
-      <artifactId>alluxio-core-server</artifactId>
+      <artifactId>alluxio-core-server-master</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.alluxio</groupId>
+      <artifactId>alluxio-core-server-worker</artifactId>
```

Comment:
I vote for following this convention also for internal dependencies as internal dependencies may also come and go, and we still shade some modules. Here is some discussion about using direct dependencies [link](http://stackoverflow.com/questions/4662011/maven-how-do-you-deal-with-dependencies-that-are-both-direct-and-transitive/4662446#4662446).

Regarding how to enforce, `mvn dependency:tree` can help. but I would create a tool or plugin to enforce it if possible.